source/reference/indexes.txt

```diff
@@ -1,278 +1,213 @@
 
-    class Person
-      include Mongoid::Document
-      field :ssn
+Declare and Create an Index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-      index({ ssn: 1 }, { unique: true, name: "ssn_index" })
-    end
+When using {+odm+}, you can declare your index using the ``index`` macro and 
+then create it using the ``create_indexes`` macro.
 
-You can define indexes on embedded document fields as well:
+The following code example shows how to declare and create an ascending index 
+named ``cuisine_index`` on the ``cuisine`` field in the ``Restaurant`` class:
 
-.. code-block:: ruby
+.. literalinclude:: /includes/reference/indexes.rb
+  :language: ruby
+  :emphasize-lines: 8, 11
+  :start-after: start create index
+  :end-before: end create index
 
-    class Person
-      include Mongoid::Document
-      embeds_many :addresses
-      index "addresses.street" => 1
-    end
+The ``index`` macro defines the index you want to create and the ``create_indexes``
+macro creates the index in the ``restaurants`` collection. 
 
-You can index on multiple fields and provide direction:
+When defining an index, the first hash object contains the field you want to
+index and its direction. ``1`` represents an ascending index, ``-1`` represents a
+descending index. The second hash object contains index options. To learn more
+about index options, see the :ref:`mongoid-indexes-api-documentation` section.
 
-.. code-block:: ruby
+Aliases and Declaring Indexes
++++++++++++++++++++++++++++++
 
-    class Person
-      include Mongoid::Document
-      field :first_name
-      field :last_name
+You can use aliased field names in index definitions. For example, the following
+example creates a field on the ``b`` field, which is an alias of the ``borough``
+field:
 
-      index({ first_name: 1, last_name: 1 }, { unique: true })
-    end
+.. literalinclude:: /includes/reference/indexes.rb
+  :language: ruby
+  :start-after: start create alias index
+  :end-before: end create alias index
 
-Indexes can be sparse:
+Create an Index on Embedded Document Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: ruby
+You can define an index on embedded document fields. The following code example
+shows how to declare an ascending index on the ``street`` field in the
+``Restaurant`` class where ``street`` is a field embedded within the ``address``
+field.
 
-    class Person
-      include Mongoid::Document
-      field :ssn
+.. literalinclude:: /includes/reference/indexes.rb
+  :language: ruby
+  :start-after: start create embedded index
+  :end-before: end create embedded index
 
-      index({ ssn: -1 }, { sparse: true })
-    end
+Create a Compound Index
+~~~~~~~~~~~~~~~~~~~~~~~
 
-For geospatial indexes, make sure the field being indexed is of type Array:
+You can define a compound index on multiple fields. The following code example
+shows how to declare a compound index that is ascending on the ``address.street``
+field and descending on the ``name`` field.
```

Comment:
[q] This description didn't seem to match the included code example which uses ``borough``. Should it be ``address.street`` in the example instead?

Reply:
oops, thanks for catching!